File: app/src/main/java/cosmic/com/mapprj/contract/LoginContract.java
package cosmic.com.mapprj.contract;

public interface LoginContract {

    interface LoginView{
        void showProgressbar();
        void hideProgressbar();
        void onSuccess();
        void onFailed(String message);

    }

    interface LoginListener{
        void onStart();

        void onSuccess();

        void onFailed(String message);
    }
}

File: app/src/main/java/cosmic/com/mapprj/contract/LoginContract.kt
package cosmic.com.mapprj.contract

interface LoginContract {

    interface LoginView {
        fun showProgressbar()
        fun hideProgressbar()
        fun onSuccess()
        fun onFailed(message: String)

    }

    interface LoginListener {
        fun onStart()
        fun onSuccess()
        fun onFailed(message: String)
    }
}
